verde/tests/test_utils.py

```diff
@@ -27,9 +27,28 @@
     meshgrid_to_1d,
     parse_engine,
     partition_by_sum,
+    fill_nans
 )
 
 
+def test_fill_nans():
+    """
+    This function tests the fill_nans function.
+    """
+
+    grid = xr.DataArray([[1, np.nan, 3],
+                        [4, 5, np.nan],
+                        [np.nan, 7, 8]])
+
+    filled_grid = fill_nans(grid)
+    expected_values = xr.DataArray([[1, 1, 3],
+                            [4, 5, 3],
+                            [4, 7, 8]])
+
```

Comment:
The `DataArray`s should contain coordinates as well. They should be as close as possible to the format of a real dataset. That's how we make the tests more robust.